src/modules/bulkServices/components/AssignServiceButton.tsx

```diff
@@ -161,4 +161,4 @@ const AssignServiceButton: React.FC<Props> = ({
   );
 };
 
-export default AssignServiceButton;
+export default memo(AssignServiceButton);
```

Comment:
> I was seeing "High memory usage" warnings from my tab locally when checking and unchecking the "select all" button on the Bulk Services page multiple times in a row. This seems to have improved it somewhat.

Hmm, memory usage warnings are concerning. Did you have react dev tools running at the time?

I'm looked a bit and I didn't see a difference in rerenders between using `memo` vs not. The buttons all _do_ re-render a lot, since the loading and disable states change as bed nights are assigned, but `memo` shouldn't affect that since those are rerenders due to prop changes. I'm sure there is something I am missing though, I didn't dig too far. 
So this seems like a fine change to me, bed night assignment still appears to work as expected!

Reply:
Yes, I did have react devtools running, maybe that was why.